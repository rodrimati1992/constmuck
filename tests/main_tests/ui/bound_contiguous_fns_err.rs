use bytemuck::Contiguous;

fn _test_from_integer<T, U>(a: T, b: T) -> (U, U) {
    (
        constmuck::contiguous::from_integer(a).unwrap(),
        Contiguous::from_integer(b).unwrap(),
    )
}

fn _test_into_integer<T, U>(a: T, b: T) -> (U, U) {
    (
        constmuck::contiguous::into_integer(a),
        Contiguous::into_integer(b),
    )
}


fn main(){}
