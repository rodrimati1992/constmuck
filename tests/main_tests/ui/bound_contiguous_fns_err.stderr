error[E0277]: the trait bound `U: Contiguous` is not satisfied
   --> tests/main_tests/ui/bound_contiguous_fns_err.rs:5:9
    |
5   |         constmuck::contiguous::from_integer(a).unwrap(),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Contiguous` is not implemented for `U`
    |
note: required by a bound in `constmuck::contiguous::from_integer`
   --> src/contiguous.rs:230:1
    |
230 | / declare_from_int_fns! {
231 | |     (from_i8,    I8,    i8)
232 | |     (from_i16,   I16,   i16)
233 | |     (from_i32,   I32,   i32)
...   |
242 | |     (from_usize, Usize, usize)
243 | | }
    | |_^ required by this bound in `from_integer`
    = note: this error originates in the macro `declare_from_int_fns` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `U`
    |
3   | fn _test_from_integer<T, U: bytemuck::Contiguous>(a: T, b: T) -> (U, U) {
    |                           ++++++++++++++++++++++

error[E0277]: the trait bound `U: Contiguous` is not satisfied
 --> tests/main_tests/ui/bound_contiguous_fns_err.rs:6:9
  |
6 |         Contiguous::from_integer(b).unwrap(),
  |         ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Contiguous` is not implemented for `U`
  |
help: consider restricting type parameter `U`
  |
3 | fn _test_from_integer<T, U: bytemuck::Contiguous>(a: T, b: T) -> (U, U) {
  |                           ++++++++++++++++++++++

error[E0277]: the trait bound `U: Contiguous` is not satisfied
 --> tests/main_tests/ui/bound_contiguous_fns_err.rs:6:9
  |
6 |         Contiguous::from_integer(b).unwrap(),
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Contiguous` is not implemented for `U`
  |
help: consider restricting type parameter `U`
  |
3 | fn _test_from_integer<T, U: bytemuck::Contiguous>(a: T, b: T) -> (U, U) {
  |                           ++++++++++++++++++++++

error[E0277]: the trait bound `U: Contiguous` is not satisfied
 --> tests/main_tests/ui/bound_contiguous_fns_err.rs:5:45
  |
5 |         constmuck::contiguous::from_integer(a).unwrap(),
  |                                             ^ the trait `Contiguous` is not implemented for `U`
  |
help: consider restricting type parameter `U`
  |
3 | fn _test_from_integer<T, U: bytemuck::Contiguous>(a: T, b: T) -> (U, U) {
  |                           ++++++++++++++++++++++

error[E0277]: the trait bound `T: Contiguous` is not satisfied
  --> tests/main_tests/ui/bound_contiguous_fns_err.rs:12:45
   |
12 |         constmuck::contiguous::into_integer(a),
   |         ----------------------------------- ^ the trait `Contiguous` is not implemented for `T`
   |         |
   |         required by a bound introduced by this call
   |
note: required by a bound in `constmuck::contiguous::into_integer`
  --> src/contiguous.rs:75:30
   |
75 | pub const fn into_integer<T: Contiguous>(value: T) -> T::Int {
   |                              ^^^^^^^^^^ required by this bound in `into_integer`
help: consider restricting type parameter `T`
   |
10 | fn _test_into_integer<T: bytemuck::Contiguous, U>(a: T, b: T) -> (U, U) {
   |                        ++++++++++++++++++++++

error[E0277]: the trait bound `T: Contiguous` is not satisfied
  --> tests/main_tests/ui/bound_contiguous_fns_err.rs:12:9
   |
12 |         constmuck::contiguous::into_integer(a),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Contiguous` is not implemented for `T`
   |
help: consider restricting type parameter `T`
   |
10 | fn _test_into_integer<T: bytemuck::Contiguous, U>(a: T, b: T) -> (U, U) {
   |                        ++++++++++++++++++++++

error[E0277]: the trait bound `T: Contiguous` is not satisfied
  --> tests/main_tests/ui/bound_contiguous_fns_err.rs:13:34
   |
13 |         Contiguous::into_integer(b),
   |         ------------------------ ^ the trait `Contiguous` is not implemented for `T`
   |         |
   |         required by a bound introduced by this call
   |
help: consider restricting type parameter `T`
   |
10 | fn _test_into_integer<T: bytemuck::Contiguous, U>(a: T, b: T) -> (U, U) {
   |                        ++++++++++++++++++++++

error[E0277]: the trait bound `T: Contiguous` is not satisfied
  --> tests/main_tests/ui/bound_contiguous_fns_err.rs:13:9
   |
13 |         Contiguous::into_integer(b),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Contiguous` is not implemented for `T`
   |
help: consider restricting type parameter `T`
   |
10 | fn _test_into_integer<T: bytemuck::Contiguous, U>(a: T, b: T) -> (U, U) {
   |                        ++++++++++++++++++++++

error[E0277]: the trait bound `T: Contiguous` is not satisfied
  --> tests/main_tests/ui/bound_contiguous_fns_err.rs:12:9
   |
12 |         constmuck::contiguous::into_integer(a),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Contiguous` is not implemented for `T`
   |
help: consider restricting type parameter `T`
   |
10 | fn _test_into_integer<T: bytemuck::Contiguous, U>(a: T, b: T) -> (U, U) {
   |                        ++++++++++++++++++++++
